BullsAndCowsTest/BullsAndCowsGameTest.cs

```diff
@@ -8,10 +9,113 @@ public class BullsAndCowsGameTest
         [Fact]
         public void Should_create_BullsAndCowsGame()
         {
-            var secretGenerator = new SecretGenerator();
+            var secretGenerator = new TestSecretGenerator();
             var game = new BullsAndCowsGame(secretGenerator);
             Assert.NotNull(game);
             Assert.True(game.CanContinue);
         }
+
+        [Fact]
+        public void Should_terminate_game_after_6_guesses()
+        {
+            var secretGenerator = new TestSecretGenerator();
+            var game = new BullsAndCowsGame(secretGenerator);
+            for (int i = 0; i < 5; i++)
+            {
+                game.Guess("5 2 3 4");
+            }
+
+            Assert.True(game.CanContinue);
+
+            game.Guess("5 2 3 4");
+            Assert.False(game.CanContinue);
+        }
+
+        [Fact]
+        public void Should_return_0A0B_when_all_wrong()
+        {
+            //given
+            var secretGenerator = new TestSecretGenerator();
+            var game = new BullsAndCowsGame(secretGenerator);
+            //when
+            string answer = game.Guess("5 6 7 8");
+            //then
+            Assert.Equal("0A0B", answer.Split(" ")[0]);
+        }
+
+        [Theory]
+        [InlineData("4 3 2 1", "4321")]
+        [InlineData("4 3 1 2", "4312")]
+        public void Should_return_4A0B_when_all_right(string guess, string secret)
+        {
+            //given
+            var mockSecretGenerator = new Mock<SecretGenerator>();
+            mockSecretGenerator.Setup(m => m.GenerateSecret()).Returns(secret);
+            var game = new BullsAndCowsGame(mockSecretGenerator.Object);
+            //when
+            string answer = game.Guess(guess);
+            //then
+            Assert.Equal("4A0B", answer.Split(" ")[0]);
+            Assert.False(game.CanContinue);
```

Comment:
a test contain pass an fail case is great

Reply:
@bukeyuanguan this one might be an issue. I was considering dividing them into separate cases. Will track this issue.